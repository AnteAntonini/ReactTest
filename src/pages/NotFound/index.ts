import { lazy } from "react";

export const NotFound = lazy(() => import("./NotFound"));
