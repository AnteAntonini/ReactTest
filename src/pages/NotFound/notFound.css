.not-found-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100vh;
  max-width: 500px;
  min-width: 300px;
  margin: 0 auto;
}

.row {
  padding-top: 2rem;
  max-width: 100%;
  width: 100%;
  display: flex;
  justify-content: center;
}

.row button {
  border-radius: 25px;
  width: 80%;
  height: 40px;
  font-size: 1.3rem;
  color: var(--color-martian-lightgray);
  font-weight: 700;
  background: var(--color-martian-red);
  border: 0px;
  cursor: pointer;
  transition: opacity 0.25s ease-out;
}

.row button:hover {
  opacity: 0.8;
}

.button {
  padding-bottom: 1.5rem;
}
