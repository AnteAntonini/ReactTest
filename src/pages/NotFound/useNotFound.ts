import { useEffect } from "react";
import { useNavigate } from "react-router-dom";

export const useNotFound = (message: string) => {
  let navigate = useNavigate();

  const handleClick = () => {
    navigate("/");
  };

  useEffect(() => {
    console.log(`${message} Not Found`);
  }, []);

  return { handleClick };
};
