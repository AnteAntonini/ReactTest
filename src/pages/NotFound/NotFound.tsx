import { useNotFound } from "./useNotFound";
import "./notFound.css";

const NotFound = ({ message }: { message: string }) => {
  const { handleClick } = useNotFound(message);

  return (
    <div className="not-found-container">
      <h2>404 | Sorry, we can't find that page.</h2>
      <div className="row ">
        <button onClick={handleClick}>Home Page</button>
      </div>
    </div>
  );
};

export default NotFound;
