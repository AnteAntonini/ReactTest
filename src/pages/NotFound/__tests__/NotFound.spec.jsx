import { render, fireEvent } from "@testing-library/react";
import NotFound from "../NotFound";
import { useNotFound } from "../useNotFound";

// Mock the useNotFound hook
jest.mock("../useNotFound");

describe("NotFound Component", () => {
  it("renders the 404 page with a button", () => {
    // Mock the useNotFound hook return values
    useNotFound.mockReturnValue({
      handleClick: jest.fn(),
    });

    const { getByText } = render(<NotFound />);

    // Check if the 404 message and button are present
    expect(
      getByText("404 | Sorry, we can't find that page.")
    ).toBeInTheDocument();
    expect(getByText("Home Page")).toBeInTheDocument();
  });

  it("calls handleClick when the button is clicked", () => {
    // Mock the useNotFound hook return values
    const mockHandleClick = jest.fn();
    useNotFound.mockReturnValue({
      handleClick: mockHandleClick,
    });

    const { getByText } = render(<NotFound />);

    const button = getByText("Home Page");

    fireEvent.click(button);

    // Verify that handleClick is called
    expect(mockHandleClick).toHaveBeenCalled();
  });
});

describe("Test snapshot", () => {
  it("Component should match snapshot", () => {
    useNotFound.mockReturnValue({
      handleClick: jest.fn(),
    });

    const { asFragment } = render(<NotFound />);

    expect(asFragment()).toMatchSnapshot();
  });
});
