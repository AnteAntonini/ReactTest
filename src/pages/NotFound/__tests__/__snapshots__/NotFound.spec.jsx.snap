// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot Component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="not-found-container"
  >
    <h2>
      404 | Sorry, we can't find that page.
    </h2>
    <div
      class="row "
    >
      <button>
        Home Page
      </button>
    </div>
  </div>
</DocumentFragment>
`;
